app/models/piece.rb

```diff
@@ -8,4 +8,91 @@ class Piece < ActiveRecord::Base
   scope :rooks, -> { where(type: 'Rook') }
   scope :queens, -> { where(type: 'Queen') }
   scope :kings, -> { where(type: 'King') }
+
+  def is_obstructed?(row, column)
+    if moving_horizontally?(row)
+      is_horizontal_obstructed?(column)
+    elsif moving_vertically?(column)
+      is_vertical_obstructed?(row)
+    elsif moving_diagonally?(row, column)
+      is_diagonal_obstructed?(row, column)
+    else
+      fail 'Input is invalid!'
+    end
+  end
+
+  private
+
+  NONE = 0
+  UP = 1
+  DOWN = -1
+  RIGHT = 1
+  LEFT = -1
+
+  def moving_horizontally?(row)
+    self.row == row
```

Comment:
love these small methods.

I think you could be even more descriptive with the variable names.

```Ruby
def moving_horizontally?(destination_row)
  row == destination_row
end
```

`self` would no longer be needed.